lib/saluki-components/src/encoders/datadog/traces/mod.rs

```diff
@@ -0,0 +1,760 @@
+        let request_builder_handle = context
+            .topology_context()
+            .global_thread_pool() // Use the shared Tokio runtime thread pool.
+            .spawn_traced_named("dd-traces-request-builder", request_builder_fut);
+
+        health.mark_ready();
+        debug!("Datadog Trace encoder started.");
+
+        loop {
+            select! {
+                biased; // makes the branches of the select statement be evaluated in order.
+
+                _ = health.live() => continue,
+                maybe_payload = payloads_rx.recv() => match maybe_payload {
+                    Some(payload) => {
+                        // Dispatch an HTTP payload to the dispatcher.
+                        if let Err(e) = context.dispatcher().dispatch(payload).await {
+                            error!("Failed to dispatch payload: {}", e);
+                        }
+                    }
+                    None => break,
+                },
+                maybe_event_buffer = context.events().next() => match maybe_event_buffer {
+                    Some(event_buffer) => events_tx.send(event_buffer).await
+                        .error_context("Failed to send event buffer to request builder task.")?,
+                    None => break,
+                },
+            }
+        }
+
+        // Drop the events sender, which signals the request builder task to stop.
+        drop(events_tx);
+
+        // Continue draining the payloads receiver until it is closed.
+        while let Some(payload) = payloads_rx.recv().await {
+            if let Err(e) = context.dispatcher().dispatch(payload).await {
+                error!("Failed to dispatch payload: {}", e);
+            }
+        }
+
+        // Request build task should now be stopped.
+        match request_builder_handle.await {
+            Ok(Ok(())) => debug!("Request builder task stopped."),
+            Ok(Err(e)) => error!(error = %e, "Request builder task failed."),
+            Err(e) => error!(error = %e, "Request builder task panicked."),
+        }
+
+        debug!("Datadog Trace encoder stopped.");
+
+        Ok(())
+    }
+}
+
+async fn run_request_builder(
+    mut trace_request_builder: RequestBuilder<TraceEndpointEncoder>, telemetry: ComponentTelemetry,
+    mut events_rx: tokio::sync::mpsc::Receiver<saluki_core::topology::EventsBuffer>,
+    payloads_tx: tokio::sync::mpsc::Sender<saluki_core::topology::PayloadsBuffer>, flush_timeout: std::time::Duration,
+) -> Result<(), GenericError> {
+    let mut pending_flush = false;
+    let pending_flush_timeout = sleep(flush_timeout);
+    tokio::pin!(pending_flush_timeout);
+
+    loop {
+        select! {
+            Some(event_buffer) = events_rx.recv() => {
+                for event in event_buffer {
+                    let trace = match event {
+                        saluki_core::data_model::event::Event::Trace(trace) => trace,
```

Comment:
Import `saluki_core::data_model::event::Event` at the top so that we don't have this fully-qualified type down here.